tests/test_snmpnotification.rb

```diff
@@ -47,27 +55,53 @@ def test_snmp_notification
     # test traps returned
     assert_equal(false, Cisco::SnmpNotification.notifications.empty?,
                  'notifications is not empty')
-    assert_equal(Cisco::SnmpNotification,
-                 Cisco::SnmpNotification.notifications['vtp notifs'].class,
-                 'vtp notifs exists')
+    if platform == :nexus
+      assert_equal(Cisco::SnmpNotification,
+                   Cisco::SnmpNotification.notifications['vtp notifs'].class,
+                   'vtp notifs exists')
+    else
+      assert_equal(Cisco::SnmpNotification,
+                   Cisco::SnmpNotification.notifications['bgp'].class,
+                   'bgp exists')
+    end
 
-    # set up some traps
-    cfs_state = Cisco::SnmpNotification.new('cfs state-change-notif')
-    bgp = Cisco::SnmpNotification.new('bgp cbgp2 state-changes')
+    if platform == :nexus
+      # set up some traps
+      cfs_state = Cisco::SnmpNotification.new('cfs state-change-notif')
+      bgp = Cisco::SnmpNotification.new('bgp cbgp2 state-changes')
 
-    # Default Checking
-    assert_equal(cfs_state.enable, false)
-    assert_equal(bgp.enable, false)
+      # Default Checking
+      assert_equal(cfs_state.enable, false)
+      assert_equal(bgp.enable, false)
 
-    cfs_state.enable = true
-    bgp.enable = true
-    assert_equal(cfs_state.enable, true)
-    assert_equal(bgp.enable, true)
+      cfs_state.enable = true
+      bgp.enable = true
+      assert_equal(cfs_state.enable, true)
+      assert_equal(bgp.enable, true)
 
-    # Setting back to default and re-checking
-    cfs_state.enable = false
-    bgp.enable = false
-    assert_equal(cfs_state.enable, false)
-    assert_equal(bgp.enable, false)
+      # Setting back to default and re-checking
+      cfs_state.enable = false
+      bgp.enable = false
+      assert_equal(cfs_state.enable, false)
+      assert_equal(bgp.enable, false)
+    else
+      # set up some traps
+      bfd = Cisco::SnmpNotification.new('bfd')
+      flash_insertion = Cisco::SnmpNotification.new('flash insertion')
+
+      # Default Checking
+      assert_equal(bfd.enable, false)
+      assert_equal(flash_insertion.enable, false)
+
+      bfd.enable = true
+      flash_insertion.enable = true
+      assert_equal(bfd.enable, true)
+      assert_equal(flash_insertion.enable, true)
+
+      # Setting back to default and re-checking
+      bfd.enable = false
+      flash_insertion.enable = false
+      assert_equal(bfd.enable, false)
```

Comment:
Also, it looks like the nexus case and XR case are almost identical code.  If you renamed the notifications to something generic (``trap1`` and ``trap2`` for example) and just passed a different value to the constructors for nexus vs. xr, would you be able to git rid of the duplication?